test/unity/toolkits/neural_net/neural_net_mocks.hpp

```diff
@@ -0,0 +1,161 @@
+    set_learning_rate_call expected_call =
+        std::move(set_learning_rate_calls_.front());
+    set_learning_rate_calls_.pop_front();
+    expected_call(lr);
+  }
+
+  float_array_map train(const float_array_map& inputs) override {
+    TS_ASSERT(!train_calls_.empty());
+    train_call expected_call = std::move(train_calls_.front());
+    train_calls_.pop_front();
+    return expected_call(inputs);
+  }
+
+  float_array_map predict(const float_array_map& inputs) const override {
+    TS_ASSERT(!predict_calls_.empty());
+    predict_call expected_call = std::move(predict_calls_.front());
+    predict_calls_.pop_front();
+    return expected_call(inputs);
+  }
+
+  float_array_map export_weights() const override {
+    return export_weights_retval_;
+  }
+
+  std::deque<set_learning_rate_call> set_learning_rate_calls_;
+  std::deque<train_call> train_calls_;
+  mutable std::deque<predict_call> predict_calls_;
+  float_array_map export_weights_retval_;
+};
+
+class mock_compute_context : public compute_context {
+ public:
+  using create_augmenter_call = std::function<std::unique_ptr<image_augmenter>(
+      const image_augmenter::options& opts)>;
+
+  using create_object_detector_call =
+      std::function<std::unique_ptr<model_backend>(
+          int n, int c_in, int h_in, int w_in, int c_out, int h_out, int w_out,
+          const float_array_map& config, const float_array_map& weights)>;
+
+  ~mock_compute_context() {
+    TS_ASSERT(create_augmenter_calls_.empty());
+    TS_ASSERT(create_object_detector_calls_.empty());
+  }
+
+  size_t memory_budget() const override { return 0; }
+
+  std::vector<std::string> gpu_names() const override { return {}; }
+
+  std::unique_ptr<image_augmenter> create_image_augmenter(
+      const image_augmenter::options& opts) override {
+    TS_ASSERT(!create_augmenter_calls_.empty());
+    create_augmenter_call expected_call =
+        std::move(create_augmenter_calls_.front());
+    create_augmenter_calls_.pop_front();
+    return expected_call(opts);
+  }
+
+  std::unique_ptr<model_backend> create_object_detector(
+      int n, int c_in, int h_in, int w_in, int c_out, int h_out, int w_out,
+      const float_array_map& config, const float_array_map& weights) override {
+    TS_ASSERT(!create_object_detector_calls_.empty());
+    create_object_detector_call expected_call =
+        std::move(create_object_detector_calls_.front());
+    create_object_detector_calls_.pop_front();
+    return expected_call(n, c_in, h_in, w_in, c_out, h_out, w_out, config,
+                         weights);
+  }
+
+  std::unique_ptr<model_backend> create_activity_classifier(
+      int n, int c_in, int h_in, int w_in, int c_out, int h_out, int w_out,
+      const float_array_map& config, const float_array_map& weights) override {
+    return nullptr;
+  }
+
+  std::unique_ptr<model_backend> create_style_transfer(
+      const float_array_map& config, const float_array_map& weights) override {
+    return nullptr;
```

Comment:
This is really helpful, thanks for doing this!

To confirm, for other toolkits, would we now add logic here to mock the respective compute contexts, instead of what we did earlier, which was: Copy over the mock_compute_context class, add logic for the toolkit in question, and return `nullptr` for all other toolkits?